File: katas/8 kyu/abbreviate-a-two-word-name/solution.js
// abbreviate-a-two-word-name
/* ------------------------------- MY SOLUTION ------------------------------ */
function abbrevName(name){
  arr = name.split(" ")
  r = []
  for (word of arr) {
    lower = word.toLowerCase()
    r.push(word.charAt(0).toUpperCase())
  }
  return r.join('.')
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function abbrevName(name){
  return name.split(' ').map(i => i[0].toUpperCase()).join('.')
}
/* ---------------------------------- TEST ---------------------------------- */
const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Basic tests", () => {
  it("Testing for fixed tests", () => {
    assert.strictEqual(abbrevName("Sam Harris"), "S.H");
    assert.strictEqual(abbrevName("Patrick Feenan"), "P.F");
    assert.strictEqual(abbrevName("Evan Cole"), "E.C");
    assert.strictEqual(abbrevName("P Favuzzi"), "P.F");
    assert.strictEqual(abbrevName("David Mendieta"), "D.M");
  });
});

File: katas/8 kyu/abbreviate-a-two-word-name/solution.py
# abbreviate-a-two-word-name
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
abbrev_name = lambda x: ".".join([x[0] , x.split(" ")[1][0]]).upper()
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def abbrevName(name):
    return '.'.join(w[0] for w in name.split()).upper()
# ----------------------------------- TEST ----------------------------------- #
@test.describe("Fixed Tests")
def basic_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(abbrev_name("Sam Harris"), "S.H")
        test.assert_equals(abbrev_name("patrick feenan"), "P.F")
        test.assert_equals(abbrev_name("Evan C"), "E.C")
        test.assert_equals(abbrev_name("P Favuzzi"), "P.F")
        test.assert_equals(abbrev_name("David Mendieta"), "D.M")